### PR TITLE
Toolchain agnostic

### PR DESCRIPTION
**Pull request by zmanian**
Simultaneous support for multirust and rustup

### CODE REVIEW

File: lib/toolchain-view.coffee [hunk 1]
```diff
@@ -3,21 +3,25 @@ _ = require 'underscore-plus'
...
 
   constructor: (serializedState) ->
     super
 
   initialize: ->
     super
-    @multirustBinPath = atom.config.get("tokamak.multirustBinPath")
+    @toolBinPath = atom.config.get("tokamak.toolBinPath")
+    configtoolchain = atom.config.get("tokamak.toolChain")
+    if configtoolchain
```
Comment (vertexclique) [inline lib/toolchain-view.coffee]: `if configtoolchain?` would be nicer, right?

File: lib/utils.coffee [hunk 2]
```diff
@@ -42,19 +43,28 @@ class Utils
...
       else
-        atom.notifications.addError("Tokamak: #{pkg} is not installed or not found in PATH",
-        {
-          detail: "If you have a #{pkg} executable, set it in токамак settings.
-          If you are sure that PATH environment variable is set and includes
-          #{pkg}, please start Atom from command line.
-          ERROR: #{data.stderrData}"
-          dismissable: true
-        })
+        if (pkg == "multirust" || pkg ="rustup") && tool_found == true
```
Comment (vertexclique) [inline lib/utils.coffee]: Is this one going to work. Btw I have run into issues after applying your patch: like couldn't change the toolchain. :)

**Comment by vertexclique** [comment]
I don't have time to test it. But seems good overall (except errors and some comments).
Can you open it against `develop` branch?
And add yourself to https://github.com/vertexclique/tokamak/blob/master/CONTRIBUTORS.md

and can you able to fix issues that we are run into when changing toolchains?

Below command won't work if we couldn't trim the last (default) some shells expect expression after all (like zsh)
`rustup default stable-x86_64-apple-darwin (default)`

Also for toolbar we need to fix this too :)
``` coffee
    @toolBar.addButton
      icon: 'tools'
      callback: 'tokamak:multirust-select-toolchain'
      tooltip: 'Change Rust Toolchain'
```

File: lib/toolchain-view.coffee [hunk 3]
```diff
@@ -3,21 +3,25 @@ _ = require 'underscore-plus'
...
+      @toolChain = configtoolchain
     @getToolchainList(@items, @toolchainExitCallback)
     @commandSubscription = atom.commands.add 'atom-workspace',
-    'tokamak:multirust-select-toolchain': => @attach()
+    'tokamak:toolBinPath-select-toolchain': => @attach()
```
Comment (vertexclique) [inline lib/toolchain-view.coffee]: This is shown in command palette as `toolbinpath select toolchain` can we do it more explanatory?

**Comment by vertexclique** [comment]
@zmanian bump?

**Comment by zmanian** [comment]
I've got sometime to work on this today.

I will 

- [ ] address your code style comments
- [ ] switch the target branch to develop
- [ ] test switching toolchains

**Comment by zmanian** [comment]
Please review. Can squash. Also open #44 as an option to merge against develop

**Pull request closed without merging**